### PR TITLE
Updates to BL5340 DVK

### PR DESCRIPTION
**Pull request by ahedin25**

### CODE REVIEW

**Review by rerickson1 — DISMISSED** [review]

File: samples/subsys/display/lvgl/boards/bl5340_dvk_cpuapp.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CONFIG_KSCAN=y
+
+CONFIG_LV_Z_POINTER_KSCAN=y
```
Comment (jfischer-no) [inline samples/subsys/display/lvgl/boards/bl5340_dvk_cpuapp.conf:3]: Please justify why this overlay is needed and can not be resolved at the board level.
You commit message is wrong.

**Review by jfischer-no — DISMISSED** [review]

File: samples/subsys/display/lvgl/boards/bl5340_dvk_cpuapp.conf (hunk 1 above)
Comment (ahedin25) [reply samples/subsys/display/lvgl/boards/bl5340_dvk_cpuapp.conf:3]: So I should add it to the Kconfig.defconfig?

**Review by rerickson1 — DISMISSED** [review]

File: boards/arm/bl5340_dvk/Kconfig.defconfig [hunk 2]
```diff
@@ -99,6 +99,53 @@ config HEAP_MEM_POOL_SIZE
 config BT_HCI_VS
 	default y if BT
 
+# BUYDISPLAY_2_8_TFT_TOUCH_ARDUINO
```
[the remark this answers is not in the document]
Comment (jfischer-no) [reply boards/arm/bl5340_dvk/Kconfig.defconfig:102]: You nailed my Friday.

**Review by rerickson1 — DISMISSED** [review]

**Comment by ahedin25** [comment]
Getting a twister failure after rebase.

> zephyr/build/zephyr/misc/generated/extra_kconfig_options.conf:3: warning: malformed string literal in assignment to BT_DEVICE_NAME (defined at subsys/bluetooth/host/Kconfig:758). Assignment ignored.

File: boards/arm/bl5340_dvk/Kconfig.defconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (jfischer-no) [reply boards/arm/bl5340_dvk/Kconfig.defconfig]: Maybe a clear description if any required at all, but not a label you just copied from the shield support.
If it is intended to be removable by the user, it should not be hardcoded into the board support in this way.

**Review by rerickson1 — DISMISSED** [review]

**Review by rerickson1 — DISMISSED** [review]

File: boards/arm/bl5340_dvk/Kconfig.defconfig [hunk 3]
```diff
@@ -83,3 +80,87 @@ endif # BOARD_BL5340_DVK_CPUAPP || BOARD_BL5340_DVK_CPUAPP_NS
...
+	default y
+
+config KSCAN
+	default y
+
+config INPUT
+	default y
+
+config LV_Z_POINTER_KSCAN
+	default y
```
Comment (nordicjm) [inline boards/arm/bl5340_dvk/Kconfig.defconfig:139]: input has replaced kscan - @fabiobaltieri for comment on this

File: boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts [hunk 4]
```diff
@@ -215,8 +219,8 @@
 		reset-gpios = <&gpio0 6 GPIO_ACTIVE_LOW>;
 		cmd-data-gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
 		rotation = <270>;
-		width = <240>;
-		height = <320>;
+		width = <320>;
+		height = <240>;
```
Comment (nordicjm) [inline boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:223]: From https://www.buydisplay.com/2-8-inch-tft-touch-shield-for-arduino-w-capacitive-touch-screen-module: `ER-TFTM028-4 is 240x320 dots 2.8" color tft lcd module display with` - has this been tested and works?

File: boards/arm/bl5340_dvk/bl5340_dvk_cpunet_common.dts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+		/* 88K */
+		slot1_partition: partition@22000 {
+			label = "image-1";
+			reg = <0x00022000 0x16000>;
+		};
+		/* 12K */
+		scratch_partition: partition@38000 {
+			label = "image-scratch";
+			reg = <0x00038000 0x3000>;
+		};
```
Comment (nordicjm) [inline boards/arm/bl5340_dvk/bl5340_dvk_cpunet_common.dts:62]: Can go

File: boards/arm/bl5340_dvk/Kconfig.defconfig (hunk 3 above)
Comment (fabiobaltieri) [reply boards/arm/bl5340_dvk/Kconfig.defconfig:139]: It's in progress, LVGL still takes kscan events. Will write an lvgl pointer driver using the input APIs down the road but for now this is still the way to go.

More importantly:

```
# 2.8" TFT Touch Shield for Arduino w/Capacitive Touch Screen Module
```

If the display is on a shield this should not go in the board dts at all. Instead the board should define an `arduino-header-r3` node and then the display can be enabled with the generic shield code that already exists in `boards/shields/buydisplay_2_8_tft_touch_arduino`.

Comment (rerickson1) [reply boards/arm/bl5340_dvk/Kconfig.defconfig:139]: Our board has the display attached already and takes work for a user to remove it. See here:
https://www.lairdconnect.com/wireless-modules/bluetooth-modules/bluetooth-5-modules/bl5340-series-multi-core-bluetooth-52-802154-nfc-modules

Comment (fabiobaltieri) [reply boards/arm/bl5340_dvk/Kconfig.defconfig:139]: Ok then this makes sense.

File: boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts (hunk 4 above)
Comment (ahedin25) [reply boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:223]: The sample app hard faults without this change.  Maybe LV_Z_POINTER_KSCAN_SWAP_XY works differently than before?

Comment (fabiobaltieri) [reply boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:223]: probably kscan pointing at the ft5336 driver that does not implement an API anymore, see my other comment, you can add a `zephyr,kscan-input` to make it work again and point `kscan_input` to that instead

**Review by fabiobaltieri — APPROVED** [review]

File: boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts (hunk 4 above)
Comment (nordicjm) [reply boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:227]: Has this been tested and working since the values are flipped?

Comment (ahedin25) [reply boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:227]: Yes. It has been tested with the lvgl sample app.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**